qemu/tests/usb_redir.py

```diff
@@ -0,0 +1,270 @@
+import logging
+import re
+import os
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import env_process
+from virttest.qemu_devices import qdevices
+from virttest.utils_params import Params
+
+from provider.storage_benchmark import generate_instance
+
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test usb redirection
+
+    1) Check host configurations
+    2) Preprocess VM
+    3) Start USB redirection via spice (optional)
+    4) Check the boot menu list (optional)
+    5) Check the redirected USB device in guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _host_config_check():
+        status = True
+        err_msg = ''
+        if option == "with_negative_config":
+            out = process.getoutput("dmesg")
+            pattern = r"usb (\d-\d(?:.\d)?):.*idVendor=%s, idProduct=%s"
+            pattern = pattern % (vendorid, productid)
+            obj = re.search(pattern, out, re.ASCII)
+            if not obj:
+                status = False
+                err_msg = "Fail to get the USB device info in host dmesg"
```

Comment:
return once status is False ?  L50, L64, L70.

Reply:
Updated, thanks!